ompi/mca/coll/han/coll_han_gather.c

```diff
@@ -432,6 +440,17 @@ mca_coll_han_gather_intra_simple(const void *sbuf, int scount,
                                    &rgap);
         tmp_buf = (char *) malloc(rsize);
         tmp_buf_start = tmp_buf - rgap;
+        if (w_rank == root && MPI_IN_PLACE == sbuf) {
```

Comment:
Same trick as before, just update the sbuf to point to the original rbuf.

Reply:
You mean set tmp_buf_start as rbuf?

Reply:
nvm I just confused myself. Let me test it and update the PR

Reply:
@bosilca the trick isn't really working, still get data corruption, is it because src_shift and dest_shift are different?

Reply:
> ... by passing MPI_IN_PLACE as the value of sendbuf at the root. In such a case, sendcount and sendtype are ignored, and the contribution of the root to the gathered vector is assumed to be already in the correct place in the receive buffer.
we need to update the sbuf to point to the location where the process data is expected to be in the rbuf, something like 
```
ptrdiff_t rextent;
ompi_datatype_type_extent(dtype, &rextent);
sbuf = rbuf + rextent * (ptrdiff_t)count * t->w_rank;
```